packages/accounts-controller/src/AccountsController.ts

```diff
@@ -776,6 +789,20 @@ export class AccountsController extends BaseController<
     });
   }
 
+  /**
+   * Retrieves the value of a specific metadata key for an existing account.
+   * @param accountId - The ID of the account.
+   * @param metadataKey - The key of the metadata to retrieve.
+   * @returns The value of the specified metadata key, or undefined if the account or metadata key does not exist.
+   */
+  #populateExistingMetadata<T extends keyof InternalAccount['metadata']>(
+    accountId: string,
```

Comment:
This could be `undefined` as well with this expression: `existingAccount?.id`?

I believe there's always a "previous account" during the update, but in terms of typing this could be `undefined` (hence the `?`)

What about new accounts, they are not used in then `updateAccounts` methods, right?